### PR TITLE
Fixes #23828 - Upgrade mocha dependency

### PR DESCRIPTION
**Pull request by ShimShtein**
mocha changed file names in `1.3.0`

### CODE REVIEW

File: bundler.d/test.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 group :test do
-  gem 'mocha', '~> 1.1'
+  gem 'mocha', '~> 1.3.0'
```
Comment (mmoll) [inline bundler.d/test.rb]: why not `~> 1.3`?

Comment (ShimShtein) [reply bundler.d/test.rb]: it was `~> 1.1` and it didn't save us the trouble. Apparently, they make changes that can break us in Y versions.

Comment (ShimShtein) [reply bundler.d/test.rb]: Maybe I'm wrong, but IMHO version freeze should make the system resilient to dependency changes, but in most unrestricting way possible. In our case, we are using features that bound to change in Y versions. Hence upgrades to the next Y should be controlled, and not sporadic.

Comment (mmoll) [reply bundler.d/test.rb]: Despite the background of this specific change: If we do pin all the dependencies we have that pesimistically we're going to have a lot of chores for all the updates.

I'd still presume dependencies to follow semver, even if some of them don't always. If we run into trouble, we're seeing it quickly and can pin more specific to fix our build and then create an issue to update the dependency.

Also, looking at https://github.com/freerange/mocha/commit/76aad9c2a9620afd0925dba381a12a3b3658745b it should probably be 1.4.

**Review by mmoll — APPROVED** [review]
I'm OK with this

**Review by mmoll — CHANGES_REQUESTED** [review]
Hang on... I actually think all is fine, also with newer mocha versions.

**Comment by ShimShtein** [comment]
@mmoll, Should be OK now.

**Comment by ShimShtein** [comment]
@mmoll should be fine now

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, thanks @ShimShtein!

**Pull request merged**